osu.Game/Rulesets/Mods/ModVisualAdjusts.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Bindables;
+using osu.Framework.Localisation;
+using osu.Game.Configuration;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.UI;
+
+namespace osu.Game.Rulesets.Mods
+{
+    public abstract class ModVisualAdjusts<T, R> : Mod, IApplicableToDrawableRuleset<T> where T : HitObject where R : DrawableRuleset<T>
+    {
+        public override string Name => "Visual Adjusts";
+        public override LocalisableString Description => "Adjust some gameplay elements that can bring some visual challenge.";
+        public override double ScoreMultiplier => 1;
+        public override string Acronym => "VA";
+        public override ModType Type => ModType.Conversion;
+
+        private void triggerAdjustsForType<B, A>(A args) where B : VisualAdjustSetting<A>
```

Comment:
This looks like pretty extreme overengineering to me. Doubly-bound generics? For what reason? There's only one usage ever, can we start simpler and see where this goes rather than lock into specific structures this early?

Also if these generic args *are* to stay, please name them better. In C# land we generally don't do the rust-like single letter generic alias thing. Convention is usually `TTypeName` (as in, a longer, *descriptive* name for the alias, preceded by a `T`).

Reply:
I would rather keep this structure with the generics in place because i already know a use case that i will soon implement (Hide the combo numbers as mentioned) this structure allows for this implementation be way easier.
Would you like me to add another usage? like remove combo colors since that is most likely easier to implement.

Reply:
The only way I can be personally convinced is if you show me at least one more usage using this exact pattern, and _also_ if you show that a naive, simple implementation using no generics is clearly inferior/much longer. If you're doing this to save 5 extra lines of code, then I argue that it's not worth the complexity overhead.

Reply:
Check latest commit, added an use case for "WhiteComboColours" setting.
But here is the comparison for generic vs non-generic.

Generic:
```cs
        private void triggerAdjustsForType<TBindable, TArgs>(TArgs args) where TBindable : VisualAdjustSetting<TArgs>
        {
            foreach (var (_, property) in this.GetOrderedSettingsSourceProperties())
            {
                if (property.GetValue(this) is TBindable bindable && bindable.Value)
                    bindable.ApplyAdjusts(args);
            }
        }

        public void ApplyToDrawableRuleset(DrawableRuleset<TObject> drawableRuleset)
        {
            triggerAdjustsForType<DrawableRulesetVisualAdjustSetting, TDrawableRuleset>((TDrawableRuleset)drawableRuleset);
        }

        public void ApplyToDrawableHitObject(DrawableHitObject drawable)
        {
            triggerAdjustsForType<DrawableHitObjectVisualAdjustSetting, DrawableHitObject<TObject>>((DrawableHitObject<TObject>)drawable);
        }
```
Non-generic:
```cs
        public void ApplyToDrawableRuleset(DrawableRuleset<TObject> drawableRuleset)
        {
            var tDrawableRuleset = (TDrawableRuleset)drawableRuleset;

            foreach (var (_, property) in this.GetOrderedSettingsSourceProperties())
            {
                if (property.GetValue(this) is DrawableRulesetVisualAdjustSetting bindable && bindable.Value)
                    bindable.ApplyAdjusts(tDrawableRuleset);
            }
        }

        public void ApplyToDrawableHitObject(DrawableHitObject drawable)
        {
            var tDrawable = (DrawableHitObject<TObject>)drawable;

            foreach (var (_, property) in this.GetOrderedSettingsSourceProperties())
            {
                if (property.GetValue(this) is DrawableHitObjectVisualAdjustSetting bindable && bindable.Value)
                    bindable.ApplyAdjusts(tDrawable);
            }
        }
```
And add another method for every new use case... the generic for me sure looks way better now.

Reply:
I was able to simplify generics usage on latest commit must be even clearer now.